huaweicloud/services/dbss/data_source_huaweicloud_dbss_operation_logs.go

```diff
@@ -0,0 +1,214 @@
+			},
+			"logs": {
+				Type:        schema.TypeList,
+				Computed:    true,
+				Description: `The list of the user operation logs.`,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"id": {
+							Type:        schema.TypeString,
+							Computed:    true,
+							Description: `The ID of the user operation log.`,
+						},
+						"name": {
+							Type:        schema.TypeString,
+							Computed:    true,
+							Description: `The name of the operation object.`,
+						},
+						"description": {
+							Type:        schema.TypeString,
+							Computed:    true,
+							Description: `The description of the user operation.`,
+						},
+						"result": {
+							Type:        schema.TypeString,
+							Computed:    true,
+							Description: `The execution result of user operation.`,
+						},
+						"action": {
+							Type:        schema.TypeString,
+							Computed:    true,
+							Description: `The type of the user operation.`,
+						},
+						"function": {
+							Type:        schema.TypeString,
+							Computed:    true,
+							Description: `The function type of the operation record.`,
+						},
+						"user": {
+							Type:        schema.TypeString,
+							Computed:    true,
+							Description: `The name of the operation user.`,
+						},
+						"time": {
+							Type:        schema.TypeString,
+							Computed:    true,
+							Description: `The time of the operation record is generated, in UTC format.`,
+						},
+					},
+				},
+			},
+		},
+	}
+}
+
+func dataSourceOperationLogsRead(_ context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	cfg := meta.(*config.Config)
+	region := cfg.GetRegion(d)
+	var (
+		logsHttpUrl = "v1/{project_id}/{instance_id}/dbss/audit/operate-log"
+		logsProduct = "dbss"
+		mErr        *multierror.Error
+	)
+
+	client, err := cfg.NewServiceClient(logsProduct, region)
+	if err != nil {
+		return diag.Errorf("error creating DBSS client: %s", err)
+	}
+
+	logsPath := client.Endpoint + logsHttpUrl
+	logsPath = strings.ReplaceAll(logsPath, "{project_id}", client.ProjectID)
+	logsPath = strings.ReplaceAll(logsPath, "{instance_id}", d.Get("instance_id").(string))
+	logsOpt := golangsdk.RequestOpts{
+		KeepResponseBody: true,
+	}
+
+	bodyParams := utils.RemoveNil(buildOperationLogsBodyParams(d))
+	logs := make([]interface{}, 0)
+	page := 1
+
+	for {
+		bodyParams["page"] = page
+		logsOpt.JSONBody = bodyParams
+		logsResp, err := client.Request("POST", logsPath, &logsOpt)
+		if err != nil {
+			return diag.FromErr(err)
+		}
+		logsRespBody, err := utils.FlattenResponse(logsResp)
+		if err != nil {
+			return diag.FromErr(err)
+		}
+		operateLog := utils.PathSearch("operate_log", logsRespBody, make([]interface{}, 0)).([]interface{})
+		logs = append(logs, operateLog...)
+
+		totalCount := utils.PathSearch("total_num", logsRespBody, float64(0))
+		if len(logs) == int(totalCount.(float64)) {
+			break
+		}
+		page++
+	}
```

Comment:
This code has the risk of an infinite loop. Why not use the response error message as a judgment condition to break out of the loop?